xarray/core/duck_array_ops.py

```diff
@@ -205,22 +206,17 @@ def _nan_argminmax_object(func, fill_value_typ, value, axis=None, **kwargs):
     return where_method(data, valid_count != 0, -1)
 
 
-def _nanmean_ddof_object(ddof, fill_value_typ, axis=None, **kwargs):
+def _nanmean_ddof_object(ddof, value, axis=None, **kwargs):
     """ In house nanmean. ddof argument will be used in _nanvar method """
     valid_count = count(value, axis=axis)
-    value = fillna(value, 0.0)
-    # TODO numpy's mean does not support object-type array, so we assume float
-    dtype = kwargs.get('dtype', None)
+    value = fillna(value, 0)
+    # As dtype inference is impossible for object dtype, we assume float
+    dtype = kwargs.pop('dtype', None)
     if dtype is None and value.dtype.kind == 'O':
         dtype = value.dtype if value.dtype.kind in ['cf'] else float
```

Comment:
Is there a good workaround to infer the output dtype of the object-typed array?
We need to pass this to dask for the next division but `dtype=object` is not allowed.

Reply:
Is this fixed by your dask PR https://github.com/dask/dask/pull/3137?

If so, can we maybe say this requires using the latest dask release?